pkg/shell/superuser.jsx

```diff
@@ -27,6 +27,12 @@ import "form-layout.scss";
 
 const _ = cockpit.gettext;
 
+export function can_do_sudo() {
+    return cockpit.spawn(["sudo", "-v", "-n"], { err: "out" })
+            .then(() => true,
+                  (err, out) => !(err.exit_status == 1 && out.match("Sorry, user .+ may not run sudo on .+\\.")));
```

Comment:
Ah, this addresses some issues above (DRY and the wrong fail/then order), but the too precise string matching and locale dependencies still remain. I. e. please run with `LC_ALL=C` and just check for `indexOf("may not run")`.

But the exit comparison seems wrong; it would now return true on a different exit code.

Reply:
I wanted this to be really specific, since having a false match will falsly prevent people from gaining root.

Reply:
OK, our tests should pick up string changes. But let's please pin the locale to C at least.

Reply:
Done.